pkg/sql/gcjob/gc_job.go

```diff
@@ -104,6 +154,23 @@ func (r schemaChangeGCResumer) Resume(ctx context.Context, execCtx interface{})
 		return err
 	}
 
+	if !progress.RangesUnsplitDone {
+		if len(details.Indexes) > 0 {
+			if err := unsplitRangesForIndexes(ctx, execCfg, details.Indexes, details.ParentID); err != nil {
+				return err
+			}
+		}
+
+		if len(details.Tables) > 0 {
+			if err := unsplitRangesForTables(ctx, execCfg, details.Tables); err != nil {
+				return err
+			}
+		}
+
+		progress.RangesUnsplitDone = true
+		persistProgress(ctx, execCfg, r.jobID, progress, runningStatusGC(progress))
+	}
```

Comment:
okay, now this is onto mega nits.  Just making this comment to share perspective, feel free to disregard. I've come to like the readability of a pattern of functions called things like:
```
func maybeUnsplitRanges(ctx context.Context, execCfg *sql.ExecutorConfig, details ...) error {
    if progress.RangesUnsplitDone {
        return nil
    }
    // ...
}
```

It's a nice conduit for commentary.